addon/ng2/tasks/git-init.js

```diff
@@ -30,6 +30,23 @@ module.exports = Task.extend({
 
     return exec('git --version')
       .then(function () {
+          
+        // "true" if cmd executes in a work tree
+        // "false if cmd executes in a '.git' tree
+        // "fatal error" if cmd executes in fresh directory (only here we should do a 'git init')  
+        return exec('git rev-parse --is-inside-work-tree')
+          .then(function () {
+            ui.writeLine(chalk.red('Directory is already under version control. Skipping initialization of git.'));
+            throw "Skipping";
+          })
+          .catch(function(error) {
+            if (error === "Skipping") {
+                throw "Skipping";
+            }
+            //ui.writeLine(chalk.green('Directory is not under version control. Initializing git...'));
+          })
+      })    
+      .then(function () {
```

Comment:
Using the two comments above, this would become:

```typescript
           .then(shouldInitGit => {
             if (shouldInitGit) {
               return exec('git init')
                 .then(...```

Makes sense?